### PR TITLE
Disaster api integration

### PR DESCRIPTION
**Pull request by DenisioMytnysiano**

### CODE REVIEW

File: backend/src/main/java/disaster/model/disasters/HazardEvent.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package disaster.model.disasters;
+
+import disaster.model.geocoding.GeocodingResult;
+import disaster.model.geocoding.SuccessGeocodingResult;
+import lombok.Data;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Data
+@Document(collection = "hazard-events")
+public class HazardEvent {
```
Comment (vvkin) [inline backend/src/main/java/disaster/model/disasters/HazardEvent.java:11]: add Id field?

File: backend/src/main/java/disaster/model/disasters/HazardEvent.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package disaster.model.disasters;
+
+import disaster.model.geocoding.GeocodingResult;
+import disaster.model.geocoding.SuccessGeocodingResult;
+import lombok.Data;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Data
+@Document(collection = "hazard-events")
```
Comment (vvkin) [inline backend/src/main/java/disaster/model/disasters/HazardEvent.java]: let's use UpperCamelCase for collection names

File: backend/src/main/java/disaster/model/geocoding/SuccessGeocodingResult.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package disaster.model.geocoding;
+
+import lombok.Data;
+
```
Comment (vvkin) [inline backend/src/main/java/disaster/model/geocoding/SuccessGeocodingResult.java:4]: run formatter pls

File: backend/src/main/java/disaster/model/mappers/OpenStreetMapGeocodeResultDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        var cityNode = addressNode.path("city");
+        if (!cityNode.isMissingNode()) {
+            name = cityNode.textValue();
+        }
+
+        var villageNode = addressNode.path("village");
+        if (!villageNode.isMissingNode()) {
+            name = villageNode.textValue();
+        }
+        //var state = addressNode.get("state").textValue();
```
Comment (vvkin) [inline backend/src/main/java/disaster/model/mappers/OpenStreetMapGeocodeResultDeserializer.java]: looks like it's not needed here

File: backend/src/main/resources/application-mongodb.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
+spring.data.mongodb.authentication-database=admin
+spring.data.mongodb.username=root
```
Comment (vvkin) [inline backend/src/main/resources/application-mongodb.properties]: save sensitive data in .env

File: backend/src/main/java/disaster/service/geocoding/OpenStreetMapGeolocationService.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    public Mono<GeocodingResult> locate(double latitude, double longitude) {
+        return webclient.get()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/reverse")
+                        .queryParam("format", "json")
+                        .queryParam("lat", latitude)
+                        .queryParam("lon", longitude)
+                        .build()
+                )
+                .exchangeToMono(response -> {
```
Comment (vvkin) [inline backend/src/main/java/disaster/service/geocoding/OpenStreetMapGeolocationService.java]: let's move error handling to `onStatus`. `exchangeToMono` can be removed in this case

File: backend/src/main/java/disaster/service/eonet/EonetHazardEventsProvider.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import reactor.core.publisher.Flux;
+
+import java.time.Duration;
+
+public class EonetHazardEventsProvider {
+    private final Flux<HazardEvent> reactiveEventProvider;
+
+    @Autowired
+    public EonetHazardEventsProvider(EonetClient client) {
+        reactiveEventProvider = Flux.interval(Duration.ofMinutes(1))
```
Comment (vvkin) [inline backend/src/main/java/disaster/service/eonet/EonetHazardEventsProvider.java]: add `Duration.ZERO` as first parameter

File: backend/src/main/java/disaster/service/eonet/EonetClient.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import disaster.model.disasters.HazardEvent;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+
+@Service
+public class EonetClient {
+
+    public static final String API_URL = "https://eonet.gsfc.nasa.gov/api/v3";
```
Comment (vvkin) [inline backend/src/main/java/disaster/service/eonet/EonetClient.java]: move to properties

File: backend/src/main/java/disaster/service/disaster_alert/DisasterAlertHazardEventsProvider.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package disaster.service.disaster_alert;
```
Comment (vvkin) [inline backend/src/main/java/disaster/service/disaster_alert/DisasterAlertHazardEventsProvider.java]: not sure that's a good idea to use underscore for the package name
https://stackoverflow.com/questions/49890803/naming-conventions-of-composed-package-names

**Review by vvkin — CHANGES_REQUESTED** [review]

File: backend/src/main/java/disaster/model/mappers/OpenStreetMapGeocodeResultDeserializer.java (hunk 4 above)
Comment (DenisioMytnysiano) [reply backend/src/main/java/disaster/model/mappers/OpenStreetMapGeocodeResultDeserializer.java]: we have to agree upon what is considered a location, a name, state, country, or the combination of them

**Pull request merged**